### ISSUE TITLE
Status of `Unifying LKL into UML` (v8, Jan 2021) ?

### ISSUE THREAD

**Issue by AkihiroSuda** [lkl/linux#513]
I still want to see the `Unifying LKL into UML` work ([v8, Jan 2021](http://lists.infradead.org/pipermail/linux-um/2021-January/001029.html)) to be merged into the upstream.

Would it be possible to see v9, or the plan was changed?

**Comment by lars18th** [comment]
Hi,

Another different, but related question: It will possible to compile UML for different architectures?

**Comment by fish4terrisa-MSDSM** [comment]
> Hi,
> 
> Another different, but related question: It will possible to compile UML for different architectures?

No, I have just tried. It just not work

**Comment by lars18th** [comment]
Hi @fish4terrisa-MSDSM ,

>> Another different, but related question: It will possible to compile UML for different architectures?

> No, I have just tried. It just not work

Me too! However, you have any idea about how to add support for any architecture different from x86 and x86_64 with a recent kernel? Can LKL help with this objective?

**Comment by fish4terrisa-MSDSM** [comment]
> Hi @fish4terrisa-MSDSM ,
> 
> > > Another different, but related question: It will possible to compile UML for different architectures?
> 
> > No, I have just tried. It just not work
> 
> Me too! However, you have any idea about how to add support for any architecture different from x86 and x86_64 with a recent kernel? Can LKL help with this objective?

I think lkl might help,but all the arch-depend codes in arch/x86/um should be replaced by lkl,as Unifying LKL into UML is not finished,it might take more time to support other archs.

**Comment by lars18th** [comment]
Hi @fish4terrisa-MSDSM ,

> I think lkl might help,but all the arch-depend codes in arch/x86/um should be replaced by lkl,as Unifying LKL into UML is not finished,it might take more time to support other archs.

From your response I feel it's potentially feasible that using LKL implementation we will obtain a _replacement_ of the code in kernel at `arch/x86/um`. That's true? If yes then I'll try to help to work on this. Any information to read first?

Regards.

**Comment by fish4terrisa-MSDSM** [comment]
> Hi @fish4terrisa-MSDSM ,
> 
> > I think lkl might help,but all the arch-depend codes in arch/x86/um should be replaced by lkl,as Unifying LKL into UML is not finished,it might take more time to support other archs.
> 
> From your response I feel it's potentially feasible that using LKL implementation we will obtain a _replacement_ of the code in kernel at `arch/x86/um`. That's true? If yes then I'll try to help to work on this. Any information to read first?
> 
> Regards.

Yes,just get a arch-independent replacement of the code in kernel at `arch/x86/um` then it will work on all platforms.
